CRM/Core/Smarty/plugins/function.help.php

```diff
@@ -45,6 +45,18 @@ function smarty_function_help($params, &$smarty) {
     $vars = $smarty->get_template_vars();
     $smarty->assign('id', $params['id'] . '-title');
 
+    // The way this works is a bit bonkers. All the .hlp files are expecting an
+    // assign called $params (which is different from our php var here called
+    // $params), and it does get assigned via ajax via
+    // CRM_Core_Page_Inline_Help when you click the help bubble (i.e. the link
+    // that we return at the bottom below). But right now when we fetch the
+    // file on the next line, there is no params. So it gives a notice. So
+    // let's assign something.
+    // It's also awkward since the ONLY reason we're fetching the file
+    // now is to get the help section's title and we don't care about the rest
+    // of the file, but that is a bit of a separate issue.
+    $smarty->assign('params', []);
```

Comment:
@demeritcowboy... This may a bit marginal, but... 

1. `{help}` appears to be a utility that's sprinkled into many different contexts. For example, `templates/CRM/Contact/Form/Merge.tpl`.
2. The name `$params` is common in Civi. For example, `Merge.tpl` has its own usage of `$params`.

I don't know if this actually a bug right now, but you probably only want to assign `params` temporarily (while processing `{help}`) rather than leaving `$params==[]` as a side-effect.

Perhaps something like this:

```php
    $smarty->pushScope(['params' => []]);
    $popScope = CRM_Utils_AutoClean::with([$smarty, 'popScope']);
```

Reply:
Or equivalently, change the idiom from

```php
$smarty->assign(VAR, VALUE);
$smarty->fetch(TEMPLATE)
```

to

```php
$smarty->fetchWith(TEMPLATE, [
  VAR => VALUE
])
```

(`fetchWith()` calls `fetch()` and temporarily assigns variables.)

Reply:
Ah I like that better, since as far as I can see popScope would still leave params at [] if there was no params before, which is often going to be the case.

And just to be extra careful, I would call fetchWith with whatever current template_vars, plus params if it's not already assigned.

Thanks.

Reply:
Oh fetchWith just does push and pop scope so it's the same thing. What we really want here is:
if params is missing, set it, then UNSET it after (only if it was missing before).
if params is there, then all good.

Reply:
Oof, I assumed that `popScope()` would unset, but I think you're right - in the case where we'd want `unset()`, it would probably leave `[]`. It's still better with this slightly-buggy cleanup (*`$params` will be restored if it's substantive...*). But yeah, we should take a stab at fixing that too...